src/coffee/directives/api/utils/fit-helper.coffee

```diff
@@ -9,5 +9,6 @@ angular.module('uiGmapgoogle-maps.directives.api.utils')
           everSet = true unless everSet
           point = if _.isFunction markerOrPoint.getPosition then markerOrPoint.getPosition() else markerOrPoint
         bounds.extend point
-      gMap.fitBounds(bounds) if everSet
+      if everSet
+        $timeout () -> gMap.fitBounds(bounds)
```

Comment:
Asynchronously execution of the fitBounds resulting on a returning value is in any case needed. I wouldn't force to use scopes inside that service. Therefore, of course, `$evalAsync` can be used. But, in the end, even the `angular.js` is using `$timeout` underneath.

In fact, given the execution of $evalAsync on $rootScope, it runs [$browser.defer](https://github.com/angular/angular.js/blob/master/src/ng/rootScope.js#L996) which pushes callbacks in a queue and [after dispatched with `setTimeout`](https://github.com/angular/angular.js/blob/master/src/ng/browser.js#L319).

I consider the one I pushed is simpler and works the same way.